packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -217,17 +226,20 @@ export class SharedTreeCore<TEditor extends ChangeFamilyEditor, TChange> extends
 	 * @param revision - The revision to associate with the change.
 	 * Defaults to a new, randomly generated, revision if not provided.
 	 */
-	protected applyChange(change: TChange, revision?: RevisionTag): void {
+	protected applyChange(change: TChange, type?: GraphCommitType, revision?: RevisionTag): void {
 		const commit = {
 			change,
 			revision: revision ?? mintRevisionTag(),
 			sessionId: this.editManager.localSessionId,
+			type,
 		};
-		const delta = this.editManager.addLocalChange(commit.revision, change, false);
-		this.transactions.repairStore?.capture(
-			this.changeFamily.intoDelta(change),
-			commit.revision,
+		const changeDelta = this.changeFamily.intoDelta(change);
+		this.undoRedoManager.trackRepairData(
+			changeDelta,
+			this.transactions.outerRevision ?? commit.revision,
 		);
```

Comment:
I wonder if it might make more sense to put this code in `submitCommit` since that's the one method that's always called when we need to inform the undo-redo manager of a new commit, and only call in that case (as opposed to `applyChange` wich is also called for changes made as part of a transaction)

Reply:
Repair data needs to be tracked before the change is applied locally or it just can't be found but the commit itself is tracked within `submitcommit`. This code has been updated to accommodate the new process for tracking repair data for transactions and I added a comment to explain this.  Lmk if you have any better ideas though, I tried moving things around and this seemed to be the only sequence that worked but I'm not super confident.